catchup/catchpointService.go

```diff
@@ -320,20 +334,37 @@ func (cs *CatchpointCatchupService) processStageLastestBlockDownload() (err erro
 		return cs.abort(fmt.Errorf("processStageLastestBlockDownload failed to retrieve catchup block round : %v", err))
 	}
 
-	fetcherFactory := MakeNetworkFetcherFactory(cs.net, 10, nil, &cs.config)
 	attemptsCount := 0
 	var blk *bookkeeping.Block
-	var client FetcherClient
 	// check to see if the current ledger might have this block. If so, we should try this first instead of downloading anything.
 	if ledgerBlock, err := cs.ledger.Block(blockRound); err == nil {
 		blk = &ledgerBlock
 	}
+
 	for {
 		attemptsCount++
 
+		peer := network.Peer(0)
+		blockDownloadDuration := time.Duration(0)
 		if blk == nil {
-			fetcher := fetcherFactory.New()
-			blk, _, client, err = fetcher.FetchBlock(cs.ctx, blockRound)
+			peer, err = cs.blocksDownloadPeerSelector.GetNextPeer()
+			if err != nil {
+				err = fmt.Errorf("processStageLastestBlockDownload: unable to obtain a list of peers to retrieve the latest block from")
+				return cs.abort(err)
+			}
+			httpPeer, validPeer := peer.(network.HTTPPeer)
+			if !validPeer {
+				cs.log.Warnf("processStageLastestBlockDownload: non-HTTP peer was provided by the peer selector")
```

Comment:
The `GetNextPeer` could return a non-HTTPPeer if the peerClass was configured with PeersConnectedOut or PeersConnectedIn. While this is not the case for the catchpointService, I think that it would be better to verify that.
( we will need this later on when integrating with the classic catchup )